spec/ics-027-interchain-account/README.md

```diff
@@ -0,0 +1,283 @@
+    onChanCloseInit,
+    onChanCloseConfirm,
+    onSendPacket,
+    onRecvPacket,
+    onTimeoutPacket,
+    onAcknowledgePacket,
+    onTimeoutPacketClose
+  })
+}
+```
+Once the `setup` function has been called, channels can be created through the IBC relayer module between instances of the interchain account module on separate chains.
+
+### **Routing module callbacks**
+
+### **Channel lifecycle management**
+
+Both machines `A` and `B` accept new channels from any module on another machine, if and only if:
+
+- The other module is bound to the "interchain account" port.
+- The channel being created is ordered.
+- The version string is empty.
+```typescript
+function onChanOpenInit(
+  order: ChannelOrder,
+  connectionHops: [Identifier],
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  counterpartyPortIdentifier: Identifier,
+  counterpartyChannelIdentifier: Identifier,
+  version: string) {
+  // only ordered channels allowed
+  abortTransactionUnless(order === ORDERED)
+  // only allow channels to "interchain-account" port on counterparty chain
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain-account")
+  // version not used at present
+  abortTransactionUnless(version === "")
+}
+```
+```typescript
+function onChanOpenTry(
+  order: ChannelOrder,
+  connectionHops: [Identifier],
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  counterpartyPortIdentifier: Identifier,
+  counterpartyChannelIdentifier: Identifier,
+  version: string,
+  counterpartyVersion: string) {
+  // only ordered channels allowed
+  abortTransactionUnless(order === ORDERED)
+  // version not used at present
+  abortTransactionUnless(version === "")
+  abortTransactionUnless(counterpartyVersion === "")
+  // only allow channels to "interchain-account" port on counterparty chain
+  abortTransactionUnless(counterpartyPortIdentifier === "interchain-account")
+}
+```
+```typescript
+function onChanOpenAck(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier,
+  version: string) {
+  // version not used at present
+  abortTransactionUnless(version === "")
+  // port has already been validated
+}
+```
+```typescript
+function onChanOpenConfirm(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // accept channel confirmations, port has already been validated
+}
+```
+```typescript
+function onChanCloseInit(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // no action necessary
+}
+```
+```typescript
+function onChanCloseConfirm(
+  portIdentifier: Identifier,
+  channelIdentifier: Identifier) {
+  // no action necessary
+}
+```
+### **Packet relay**
+
+In plain English, between chains `A` and `B`. It will describe only the case that chain A wants to register an Interchain account on chain B and control it. Moreover, this system can also be applied the other way around. In these set of examples, we follow Cosmos-SDK's modularity rule and its `msg` system.
```

Comment:
I am not a huge fan of using Cosmos SDK specific abstractions. Can we just leave it at the level of "tx bytes"?